File: src/main/java/test/MergeSortTest.java
package main.java.test;

import main.java.sort.MergeSort;

import java.util.Arrays;

import static main.java.utils.SortUtil.getRandomArray;
import static main.java.utils.SortUtil.printSortTime;

/**
 * Description : MergeSort测试用例
 * <p>
 * <p>Author : Matrix [xhyrzldf@foxmail.com]
 * <p>
 * <p>Date : 2017/10/7 18:17
 */
public class MergeSortTest {
    public static void main(String[] args) {
        int n = 150000;
        int[] arr = getRandomArray(n, 150000);
        int[] arr02 = Arrays.copyOf(arr, n);
        int[] arr03 = Arrays.copyOf(arr, n);
        printSortTime(arr, MergeSort::mergeSort, "mergeSort");
        printSortTime(arr02, MergeSort::mergeSortBU, "mergeSortBottomUp");
        printSortTime(arr03, Arrays::sort, "jdkSort");
    }
}

File: src/main/java/test/MergeSortTest.kt
package test

import sort.MergeSort
import utils.SortUtil.getRandomArray
import utils.SortUtil.printSortTime
import java.util.*
import java.util.function.Consumer

/**
 * Description : MergeSort测试用例
 *
 *
 *
 * Author : Matrix [xhyrzldf@foxmail.com]
 *
 *
 *
 * Date : 2017/10/7 18:17
 */
object MergeSortTest {
    @JvmStatic
    fun main(args: Array<String>) {
        val n = 150000
        val arr = getRandomArray(n, 150000)
        val arr02 = Arrays.copyOf(arr, n)
        val arr03 = Arrays.copyOf(arr, n)
        printSortTime(arr, Consumer<IntArray> { MergeSort.mergeSort(it) }, "mergeSort")
        printSortTime(arr02, Consumer<IntArray> { MergeSort.mergeSortBU(it) }, "mergeSortBottomUp")
        printSortTime(arr03, Consumer<IntArray> { Arrays.sort(it) }, "jdkSort")
    }
}
